napi_js_props.md

```diff
@@ -0,0 +1,225 @@
+# N-API - JavaScript Data Types and Values 
+
+N-API exposes a set of APIs to interrogate and set properties on JavaScript objects.
+Some of these types are documented under [Section 7](https://tc39.github.io/ecma262/#sec-operations-on-objects) 
+of the [ECMAScript Language Specification](https://tc39.github.io/ecma262/).
+
+Fundamentally, all properties in N-API can be represented in one of the following forms:
+- Named: a simple utf8 encoded string
+- Integer-Indexed: an index value represented by `uint32_t`
+- JavaScript Object: a complex JavaScript Object represented by `napi_value`
+
+N-API values are represented by the type `napi_value`.
+Any N-API API that requires a JavaScript object takes in a `napi_value`
+however, it's the caller's responsibility to make sure that the 
+`napi_value` in question is of the JavaScript type expected by the API.
+
+
+## Functions
+
+### *napi_get_property_names*
+
+#### Signature
+```
+napi_status napi_get_property_names(napi_env env,
+                                    napi_value object,
+                                    napi_value* result);
+```
+
+#### Parameters
+- `[in]  env`: The environment that the N-API API is invoked under
+- `[in]  object`: The object from which to retrieve the properties
+- `[out] result`: A `napi_value` representing an arrray of JavaScript objects that represent the property names of the object
```

Comment:
I would say "a &#96;napi_value&#96; representing a JavaScript array containing the property names of the object" in order to emphasize that the `napi_value` contains a JavaScript array that needs to be accessed using `napi_get_element()`. In fact, you should probably add, "You can iterate over &#96;result&#96; using &#96;napi_get_array_length()&#96; and &#96;napi_get_element()&#96;."